File: testserver/src/test/java/io/lepo/lukki/testserver/PageHtmlFormatterTests.java
package io.lepo.lukki.testserver;

import java.util.Arrays;
import java.util.Collections;
import java.util.stream.Stream;
import org.assertj.core.api.Assertions;
import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

class PageHtmlFormatterTests {

  @Test
  @DisplayName("Generated HTML should contain the given links")
  void testLinksInHtml() {
    String[] links = {"/foo", "/bar", "/baz"};

    String html = PageHtmlFormatter.format("/mypage", Arrays.asList(links));
    Document htmlDoc = Jsoup.parse(html);
    Stream<String> hrefs = htmlDoc.select("a").stream().map((a) -> a.attr("href"));

    Assertions.assertThat(hrefs).containsExactly(links);
  }

  @Test
  @DisplayName("Generated HTML should contain the page link")
  void testPageLinkInHtml() {
    String pageLink = "/mypage";
    String html = PageHtmlFormatter.format(pageLink, Collections.emptyList());

    Document htmlDoc = Jsoup.parse(html);

    Assertions.assertThat(htmlDoc.title()).isEqualTo(pageLink);
  }
}

File: testserver/src/test/kotlin/io/lepo/lukki/testserver/PageHtmlFormatterTests.kt
package io.lepo.lukki.testserver

import org.assertj.core.api.Assertions
import org.jsoup.Jsoup
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test
import java.util.*

internal class PageHtmlFormatterTests {

    @Test
    @DisplayName("Generated HTML should contain the given links")
    fun testLinksInHtml() {
        val links = arrayOf("/foo", "/bar", "/baz")

        val html = PageHtmlFormatter.format("/mypage", Arrays.asList(*links))
        val htmlDoc = Jsoup.parse(html)
        val hrefs = htmlDoc.select("a").map { it.attr("href") }

        Assertions.assertThat(hrefs).containsExactly(*links)
    }

    @Test
    @DisplayName("Generated HTML should contain the page link")
    fun testPageLinkInHtml() {
        val pageLink = "/mypage"
        val html = PageHtmlFormatter.format(pageLink, emptyList())

        val htmlDoc = Jsoup.parse(html)

        Assertions.assertThat(htmlDoc.title()).isEqualTo(pageLink)
    }
}
